### PR TITLE
@W-16176547: [MSDK Android] Code Review Follow-Up For W-15993541/W-15993636 - AILTN Publish & Push Notification Registration Updates

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🎸 _Ready For Review_ 🥁

  This applies some code review feedback as a fast-follow to #2587.  The crux change is switching both tasks to a few enum types.

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java [hunk 1]
```diff
@@ -564,4 +541,29 @@ private static void recreateAnalyticsPeriodicBackgroundPublishingWorkRequest() {
                 (long) publishPeriodicallyFrequencyHours
         );
     }
+
+    /**
+     * The available Salesforce analytics publishing types.
+     */
+    public enum SalesforceAnalyticsPublishingType {
```
Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java:548]: Maybe the values could be:
- PublishDisabled
- PublishOnAppBackground
- PublishPeriodically

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt [hunk 2]
```diff
@@ -531,4 +511,25 @@ open class PushService {
             }
         }
     }
+
+    enum class PushNotificationRegistrationType {
```
Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt:515]: Should the class name say ReRegistration ie be PushNotificationReRegistrationType?
Should the values "simply" be:
- ReRegistrationDisabled
- ReRegistrationOnAppForeground
- ReRegistrationPeriodically

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java (hunk 1 above)
Comment (JohnsonEricAtSalesforce) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java:550]: You got it 💫

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt (hunk 2 above)
Comment (JohnsonEricAtSalesforce) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt]: Updated!  Badda-bing, badda boom 🚀

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/AnalyticsPublishingWorker.kt [hunk 3]
```diff
@@ -98,27 +100,27 @@ internal class AnalyticsPublishingWorker(
...
+         * periodic background publishing in hours
          * @return UUID The worker's unique id, which may be used for
          * cancellation
          */
         fun enqueueAnalyticsPublishWorkRequest(
             context: Context,
-            periodicBackgroundPublishingHoursInterval: Long? = null
-        ) = when (periodicBackgroundPublishingHoursInterval) {
+            periodicBackgroundPublishingHoursInterval: Long = SalesforceAnalyticsManager.getPublishPeriodicallyFrequencyHours().toLong()
+        ): UUID? = when (SalesforceAnalyticsManager.analyticsPublishingType()) {
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/AnalyticsPublishingWorker.kt:119]: The changes in this source all reflect the switch to the new enum types.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java [hunk 4]
```diff
@@ -564,4 +543,29 @@ private static void recreateAnalyticsPeriodicBackgroundPublishingWorkRequest() {
...
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java:550]: The changes here are just for the new enum types.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.kt [hunk 5]
```diff
@@ -905,7 +908,7 @@ open class SalesforceSDKManager protected constructor(
             account = account,
             frontActivity = frontActivity,
             showLoginPage = showLoginPage,
-            reason = LogoutReason.UNKNOWN
+            reason = UNKNOWN
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.kt:911]: Slightly unrelated, but I did notice in the diff for the previous pull request I forgot this static import and that's always a nice way to reduce visual clutter.  I could undo this if we wish.

**Review by wmathurin — APPROVED** [review]

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushMessaging.kt [hunk 6]
```diff
@@ -473,12 +474,14 @@ object PushMessaging {
             enqueuePushNotificationsRegistrationWork(
                 userAccount = null,
                 action = action,
+                pushNotificationsRegistrationType = ReRegistrationOnAppForeground,
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushMessaging.kt:477]: I did find one logic fix to make from the previous pull request.  Regardless of the app's chosen push notification re-registration type, push notification registration has to happen once when the token is obtained.  So, I added an override in this call to allow that.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt [hunk 7]
```diff
@@ -246,7 +250,7 @@ open class PushService {
             getRestClient(account)?.let { restClient ->
                 val apiVersion = ApiVersionStrings.getVersionNumber(SalesforceSDKManager.getInstance().appContext)
                 // TODO remove once MSDK default api version is 61 or greater
-                if (apiVersion.compareTo("v61.0") >= 0) {
+                if (apiVersion >= "v61.0") {
```
Comment (JohnsonEricAtSalesforce) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt:253]: This was the last inspector warning in this source, so I went in for a little clean up.

**Review by JohnsonEricAtSalesforce — COMMENTED** [review]
Now this is ready for review after some testing, a little more cleanup and a quick fix to ensure push notifications registration occurs each time it's needed.

**Review by brandonpage — APPROVED** [review]

**Pull request merged**